packages/react-next/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -1,264 +1,190 @@
+    }),
+    [inputElementRef, hasFocus],
+  );
+};
+
+export const SearchBoxBase = React.forwardRef((props: ISearchBoxProps, forwardedRef: React.Ref<HTMLDivElement>) => {
+  const [hasFocus, { toggle: toggleHasFocus }] = useBoolean(false);
+  // tslint:disable-next-line:prefer-const deprecation
+  let [value, setValue] = useControllableValue(props.value, props.defaultValue, props.onChange);
+  const rootElementRef = React.useRef<HTMLDivElement>(null);
+  const inputElementRef = React.useRef<HTMLInputElement>(null);
+  const mergedRootRef = useMergedRefs(rootElementRef, forwardedRef);
+  const fallbackId = useId(COMPONENT_NAME);
+
+  // Ensure value is always a string-friendly type.
+  if (value === null || value === undefined) {
+    value = '';
   }
 
-  private _onClear(ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement> | React.KeyboardEvent<HTMLElement>) {
-    this.props.onClear && this.props.onClear(ev);
+  const {
+    ariaLabel,
+    placeholder,
+    className,
+    disabled,
+    underlined,
+    styles,
+    /* tslint:disable-next-line:deprecation */
+    labelText,
+    theme,
+    clearButtonProps = { ariaLabel: 'Clear text' },
+    disableAnimation = false,
+    iconProps,
+    id = fallbackId,
+  } = props;
+
+  const classNames = getClassNames(styles!, {
+    theme: theme!,
+    className,
+    underlined,
+    hasFocus,
+    disabled,
+    hasInput: value!.length > 0,
+    disableAnimation,
+  });
+
+  const nativeProps = getNativeProps<React.InputHTMLAttributes<HTMLInputElement>>(props, inputProperties, [
+    'className',
+    'placeholder',
+    'onFocus',
+    'onBlur',
+    'value',
+  ]);
+
+  const placeholderValue = placeholder !== undefined ? placeholder : labelText;
+
+  const onClear = (ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement> | React.KeyboardEvent<HTMLElement>) => {
+    props.onClear && props.onClear(ev);
     if (!ev.defaultPrevented) {
-      this._latestValue = '';
-      this.setState({
-        value: '',
-      });
-      this._callOnChange(undefined, '');
+      setValue('');
+      inputElementRef.current?.focus();
       ev.stopPropagation();
       ev.preventDefault();
-
-      this.focus();
     }
-  }
+  };
 
-  private _onClickFocus = () => {
-    const inputElement = this._inputElement.current;
-    if (inputElement) {
-      this.focus();
-      inputElement.selectionStart = inputElement.selectionEnd = 0;
+  const onClearClick = (ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>) => {
+    if (clearButtonProps && clearButtonProps.onClick) {
+      clearButtonProps.onClick(ev);
+    }
+    if (!ev.defaultPrevented) {
+      onClear(ev);
     }
   };
 
-  private _onFocusCapture = (ev: React.FocusEvent<HTMLElement>) => {
-    this.setState({
-      hasFocus: true,
-    });
-
-    if (this.props.onFocus) {
-      this.props.onFocus(ev as React.FocusEvent<HTMLInputElement>);
+  const onFocusCapture = (ev: React.FocusEvent<HTMLElement>) => {
+    toggleHasFocus();
```

Comment:
hmm you toggle the `hasFocus` here and other places. but previously we are explicitly setting to true/false.

1. it might work, idk. logic wise it seems risky
2. it's less readable

let's call `setTrue` or `setFalse` instead pls